src/components/organisms/UiContentArea/UiContentArea.vue

```diff
@@ -0,0 +1,293 @@
+                :key="key"
+              >
+                <UiContentAreaItem
+                  :label="item.label"
+                  :title="item.title"
+                  :name="item.name"
+                >
+                  <slot
+                    v-bind="{ item }"
+                    :name="item.name"
+                  />
+                </UiContentAreaItem>
+              </template>
+            </slot>
+          </div>
+        </slot>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import {
+  ref,
+  computed,
+  provide,
+  inject,
+} from 'vue';
+import type { PropType } from 'vue';
+import type { PropsAttrs } from '@/types/attrs';
+import type { Icon } from '../../../types/icon';
+import type { MenuSuffixVisible } from '../UiMenu/UiMenu.vue';
+import UiButton from '../../atoms/UiButton/UiButton.vue';
+import UiIcon from '../../atoms/UiIcon/UiIcon.vue';
+import UiHeading from '../../atoms/UiHeading/UiHeading.vue';
+import UiMenu from '../UiMenu/UiMenu.vue';
+import UiContentAreaItem from './_internal/UiContentAreaItem.vue';
+
+export interface ContentAreaItem {
+  label: string;
+  title: string;
+  name?: string;
+  [key: string]: unknown;
+}
+
+const props = defineProps({
+  modelValue: {
+    type: [
+      Array,
+      Object,
+    ] as PropType<ContentAreaItem[] | ContentAreaItem>,
+    default: () => ([]),
+  },
+  /**
+   * Use this props to set inside pages title.
+   */
+  title: {
+    type: String,
+    default: '',
+  },
+  /**
+   * Use this props to pass inside pages items.
+   */
+  items: {
+    type: Array as PropType<ContentAreaItem[]>,
+    default: () => ([]),
+  },
+  /**
+   * Use this props to display inside pages header.
+   */
+  hasHeader: {
+    type: Boolean,
+    default: true,
+  },
+  /**
+   * Use this props to pass attrs for back UiButton
+   */
+  buttonBackAttrs: {
+    type: Object as PropsAttrs,
+    default: () => ({}),
+  },
+  /**
+   * Use this props to pass attrs for back button UiIcon
+   */
+  iconBackAttrs: {
+    type: Object as PropsAttrs,
+    default: () => ({ icon: 'chevron-left' }),
+  },
+  /**
+   * Use this props to pass attrs for title UiHeading
+   */
+  headingTitleAttrs: {
+    type: Object as PropsAttrs,
+    default: () => ({}),
+  },
+  /**
+   * Use this props to pass labels inside component translation.
+   */
+  translation: {
+    type: Object,
+    default: () => ({ back: 'Back to' }),
+  },
+});
+const defaultProps = computed(() => ({
+  translation: {
+    back: 'Back to',
+    ...props.translation,
+  },
+  iconBackAttrs: {
+    icon: 'chevron-left' as Icon,
+    ...props.iconBackAttrs,
+  },
+}
+));
+
+const emit = defineEmits<{(event: 'update:modelValue', value: ContentAreaItem[]): void;
+}>();
+
+const index = inject('index', 0);
+provide('index', index + 1);
+const isNested = computed(() => index > 0);
+const headerIsDisplayed = computed(() => (props.hasHeader && !isNested.value));
```

Comment:
I don't know about your habits, but I see other `booleans` variables are named with regarding the _**Answering to the question**_ pattern, so WDYT about renaming this `headerIsDisplayed` to `isHeaderDisplayed`? :)